src/shared-props.ts

```diff
@@ -0,0 +1,170 @@
+import type { Prop } from 'vue';
+import { getLibrarySettings } from '@/helpers';
+import type { PropType } from 'vue';
+import type { StyleType } from 'types';
+import { styleTypes } from 'types';
+
+/**
+ * Boolean flag to indicate if the component
+ * should display its loading state.
+ *
+ * @default false
+ */
+export const loading: Prop<boolean> = {
+    type: Boolean,
+    default: false
+};
+
+/**
+ * The input label.
+ */
+export const label: Prop<string> = {
+    type: String
+};
+
+/**
+ * The input's disabled state.
+ *
+ * @default false
+ */
+export const disabled: Prop<boolean, false> = {
+    type: Boolean,
+    default: false
+};
+
+/**
+ * The input name.
+ */
+export const name: Prop<string> = {
+    type: String,
+    required: true
+};
+
+/**
+ * Replaces the standard HTML autofocus,
+ * which only works on page load.
+ *
+ * @default false
+ */
+export const autofocus: Prop<boolean, false> = {
+    type: Boolean,
+    default: false
+};
+
+/**
+ * The placeholder the input should display.
+ */
+export const placeholder: Prop<string> = {
+    type: String
+};
+
+/**
+ * String to prefix the input with, eg: £.
+ */
+export const prefix: Prop<string> = {
+    type: String
+};
+
+/**
+ * String to suffix the input with, eg: £.
+ */
+export const suffix: Prop<string> = {
+    type: String
+};
```

Comment:
Non-required, consider a default for consistancy (I know the linting doesn't kick in here)